include/PathSimulator.hpp

```diff
@@ -0,0 +1,140 @@
+        };
+
+        using Path  = std::vector<std::pair<std::size_t, std::size_t>>;
+        using Steps = std::vector<Step>;
+
+        SimulationPath() = default;
+        SimulationPath(std::size_t nleaves, Path path, const qc::QuantumComputation* qc, bool assumeCorrectOrder = false);
+
+        Path                          path{};
+        Steps                         steps{};
+        std::size_t                   nleaves{};
+        const qc::QuantumComputation* qc{};
+    };
+
+    explicit PathSimulator(std::unique_ptr<qc::QuantumComputation>&& qc, Mode mode = Mode::Sequential, std::size_t nthreads = std::thread::hardware_concurrency()):
+        CircuitSimulator(std::move(qc)), executor(nthreads) {
+        // remove final measurements implement measurement support for task-based simulation
+        //        qc::CircuitOptimizer::removeFinalMeasurements(*(this->qc));
+        // Add new strategies here
+        switch (mode) {
+            case Mode::BracketGrouping3:
+                generateBracketSimulationPath(3);
+                break;
+            case Mode::PairwiseRecursiveGrouping:
+                generatePairwiseRecursiveGroupingSimulationPath();
+                break;
+            case Mode::SequentialFalse:
+                generateSequentialSimulationPathFalseOrder();
+                break;
+            case Mode::Cotengra:
+            case Mode::AvgCase:
+            case Mode::BestCase:
+                // in this case the contraction plan is explicitly set
+                break;
+            case Mode::Sequential:
+            default:
+                generateSequentialSimulationPath();
+                break;
+        }
+    }
+
+    PathSimulator(std::unique_ptr<qc::QuantumComputation>&& qc, const ApproximationInfo approx_info, const unsigned long long seed, Mode mode = Mode::Sequential, const std::size_t nthreads = 1):
+        CircuitSimulator(std::move(qc), approx_info, seed), executor(nthreads) {
+        // remove final measurements implement measurement support for task-based simulation
+        //      qc::CircuitOptimizer::removeFinalMeasurements(*(this->qc));
+        // Add new strategies here
+        switch (mode) {
+            case Mode::BracketGrouping3:
+                generateBracketSimulationPath(3);
+                break;
+            case Mode::PairwiseRecursiveGrouping:
+                generatePairwiseRecursiveGroupingSimulationPath();
+                break;
+            case Mode::SequentialFalse:
+                generateSequentialSimulationPathFalseOrder();
+                break;
+            case Mode::Cotengra:
+            case Mode::AvgCase:
+            case Mode::BestCase:
+                // in this case the contraction plan is explicitly set
+                break;
+            case Mode::Sequential:
+            default:
+                generateSequentialSimulationPath();
+                break;
+        }
+    }
```

Comment:
Since we do not support ApproximationInfo in this simulator, it is probably best to just combine both constructors into one that takes a `std::size_t seed = 0` as its last argument.
This safes duplicate code and also simplifies the tests

Reply:
Added the bracket size as an additional parameter